configure

```diff
@@ -801,11 +801,29 @@ echo "ABI_FLAG=$ABI_FLAG" >> Makefile
 echo "PIC_FLAG=$PIC_FLAG" >> Makefile
 echo "EXTRA_SHARED_FLAGS=$EXTRA_SHARED_FLAGS" >> Makefile
 echo "" >> Makefile
-echo "DLPATH=$DLPATH" >> Makefile
-echo "DLPATH_ADD=$DLPATH_ADD" >> Makefile
 echo "EXTENSIONS=$EXTENSIONS" >> Makefile
 echo "EXTRA_BUILD_DIRS=$EXTRA_BUILD" >> Makefile
 echo "" >> Makefile
+case "$OS" in
+    MINGW*|CYGWIN*)
+        echo "LDD=echo" >> Makefile;;
+    Darwin)
+        echo "LDD=otool -L" >> Makefile;;
+    *)
+        echo "LDD=ldd" >> Makefile;;
+esac;
+case "$OS" in
+    MINGW*)
+        echo "DLPATH=$DLPATH" >> Makefile
+        echo "DLPATH_ADD=$DLPATH_ADD" >> Makefile
+        echo 'ifdef $(DLPATH)' >> Makefile
+        echo '    $(DLPATH):=$($(DLPATH)):$(DLPATH_ADD)' >> Makefile
+        echo 'else' >> Makefile
+        echo '    $(DLPATH):=$(DLPATH_ADD)' >> Makefile
+        echo 'endif' >> Makefile;;
+    *)
+        echo 'check: LDFLAGS := -Wl,-rpath,$(shell pwd) $(LDFLAGS)' >> Makefile;;
```

Comment:
`DLPATH_ADD` used `${LIB_DIRS} ${EXTRA_LIB_DIRS}` maybe they need to be added here as well.